### PR TITLE
Add local config sample

### PR DESCRIPTION
**Pull request by nullvariable**
Co-authored-by: [dylanjameswagner](https://github.com/dylanjameswagner) 
Co-authored-by: [Doug Cone](https://github.com/nullvariable)

Replaces #211 with fresh code, props to Dylan James Wagner for the original PR.
Fixes: #178 
Fixes: #59

### CODE REVIEW

File: wp-config-local-sample.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * This is a sample config for local development. wp-config.php will
+ * load this file if you're not in a Pantheon environment. Simply edit/copy
+ * as needed and rename to wp-config-local.php.
+ *
```
[the remark this answers is not in the document]
Comment (nullvariable) [reply wp-config-local-sample.php:6]: that adds some maintenance overhead of ensuring that link is still correct. Do we have a process that we can follow or a CI script we can leverage to ensure it's flagged if that url changes?

[the remark this answers is not in the document]
Comment (pwtyler) [reply wp-config-local-sample.php:6]: Perhaps simply a link in the other direction then. Currently docs point to [wp-config.php](https://github.com/pantheon-systems/WordPress/blob/master/wp-config.php#L11-L20), but we could open a PR to point to this file instead once merged to default/master.

**Comment by nullvariable** [comment]
@pwtyler we should roll this into your `wp-config` changes too

**Comment by greg-1-anderson** [comment]
Maybe all of these sample lines should be commented out. That's more editing for folks who want to use the file, but safer if someone includes it by accident. (Maybe they deserve what they get if they do that tho?)

Usually we use "example" rather than "sample". I don't know if "sample" is more common in the WordPress domain.

**Comment by pwtyler** [comment]
> Maybe all of these sample lines should be commented out. That's more editing for folks who want to use the file, but safer if someone includes it by accident. (Maybe they deserve what they get if they do that tho?)

The user would already have to change the file from `wp-config-local-sample.php` to `wp-config-local.php`, so I wouldn't be too worried, and this follows the model of `wp-config-sample.php` (already part of WordPress core), which has the same default values as used here.

**Comment by dylanjameswagner** [comment]
@nullvariable In my PR #211 I included whitespace between the constant name and value which aligns all the values in each section for readability. There is precedent for this alignment in the WordPress core file `wp-config-sample.php`, specifically with the keys and salts. I find that way far easier to scan and understand. **I suggest this whitespace be included.**

Additionally, in PR #211 the constants `WP_HOME` and `WP_SITEURL` are being set using `$_SERVER['HTTP_HOST']` which relieves the user from having to set a specific URL for their local environment which is handy for using development tools that change the URL for specific tasks/workflows like compiling tools such as Live Reload or BrowserSync, proxy URLs, xip.io URLs, or local network share URLs. **I suggest using this technique for those values.**

**Pull request merged**